src/client/Microsoft.Identity.Client/MsalThrottledUiRequiredException.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Exception type thrown when MSAL detects that an application is trying to acquire a token even 
+    /// though an <see cref="MsalUiRequiredException"/> was recently thrown. 
+    /// To mitigate this, when an <see cref="MsalUiRequiredException"/> is encountered,
+    /// the application should switch to acquiring a token interactively. To better understand
+    /// why the <see cref="MsalUiRequiredException" /> was thrown, inspect the <see cref="MsalUiRequiredException.Classification"/>
+    /// property.
+    /// 
+    /// The properties of this exception are identical to the original exception
+    /// 
+    /// For more details see https://aka.ms/msal-net-throttling
+    /// </summary>
+    public class MsalThrottledUiRequiredException : MsalUiRequiredException
```

Comment:
@rayluo - not sure if the spec makes it clear, but @SomkaPe argued for having explicit exceptions. In java / .net we used inheritance to not break people. In python it's up to you, but it should be clear that throttling occurs otherwise it will be difficult to fix support cases etc.